### DISCUSSION TITLE
Filtering on a child node object (not a collection)

### DISCUSSION THREAD

**Discussion by bo55vxr** [ChilliCream/hotchocolate#5101] [Q&A] [answered]
Assume this (pseudo) definition

```
Person
  First Name
  Last Name
  CurrentRole
    RoleName
    RoleCode   
```
and I have an operation called `GetPeople` (with no parameters)

If the client app wanted to filter the response for a specific `RoleCode` how would they construct the query.

I have tried this but it is not working

```
query
{
  people()
    First Name
  Last Name
  CurrentRole (where: {RoleCode: {eq: "XX"}})
    RoleName
    RoleCode
```

Any suggestions please

**Comment by arjundnair-fingent** [comment, marked as answer]
If the child node is not a collection there will be only one role, right? That role will have a single role code. If you want to filter the `Person` object using `RoleCode`, I'd suggest you to add the where block on the `Person` resolver like this:

```
query {
  person(where: { currentrole: { RoleCode: { eq: "XX" } } }) {
    firstname
    lastname
    currentrole {
      rolename
      rolecode
    }
  }
}
```

**Reply by bo55vxr** [reply]
Smashing ... Thanks
